pkg/driver/openstack-cinder/openstack_cinder.go

```diff
@@ -0,0 +1,205 @@
+		klog.Error(nil, "Flavour HyperShift is not supported!")
+		return nil, fmt.Errorf("Flavour HyperShift is not supported!")
+	}
+
+	cfg := operator.NewDefaultOperatorControllerConfig(flavour, c, "OpenStackCinder")
+
+	go c.ConfigInformers.Start(ctx.Done())
+
+	// Hooks to run on all clusters
+	cfg.AddDeploymentHookBuilders(c, withCABundleDeploymentHook, withConfigDeploymentHook)
+	cfg.AddDaemonSetHookBuilders(c, withCABundleDaemonSetHook, withConfigDaemonSetHook)
+
+	cfg.DeploymentWatchedSecretNames = append(cfg.DeploymentWatchedSecretNames, cloudCredSecretName, metricsCertSecretName)
+
+	configMapSyncer, err := createConfigMapSyncer(c)
+	if err != nil {
+		return nil, err
+	}
+	cfg.ExtraControlPlaneControllers = append(cfg.ExtraControlPlaneControllers, configMapSyncer)
+
+	return cfg, nil
+}
+
+// withCABundleDeploymentHook projects custom CA bundle ConfigMap into the CSI driver container
+func withCABundleDeploymentHook(c *clients.Clients) (dc.DeploymentHookFunc, []factory.Informer) {
+	hook := csidrivercontrollerservicecontroller.WithCABundleDeploymentHook(
+		c.ControlPlaneNamespace,
+		trustedCAConfigMap,
+		c.GetControlPlaneConfigMapInformer(c.ControlPlaneNamespace),
+	)
+	informers := []factory.Informer{
+		c.GetControlPlaneConfigMapInformer(c.ControlPlaneNamespace).Informer(),
+	}
+	return hook, informers
+}
+
+// withCABundleDaemonSetHook projects custom CA bundle ConfigMap into the CSI driver container
+func withCABundleDaemonSetHook(c *clients.Clients) (csidrivernodeservicecontroller.DaemonSetHookFunc, []factory.Informer) {
+	hook := csidrivernodeservicecontroller.WithCABundleDaemonSetHook(
+		clients.CSIDriverNamespace,
+		trustedCAConfigMap,
+		c.GetConfigMapInformer(clients.CSIDriverNamespace),
+	)
+	informers := []factory.Informer{
+		c.GetConfigMapInformer(clients.CSIDriverNamespace).Informer(),
+	}
+	return hook, informers
+}
+
+// withConfigDeploymentHook adds annotations based on the hash of the config map containing our config,
+// ensuring we restart if that changes
+func withConfigDeploymentHook(c *clients.Clients) (dc.DeploymentHookFunc, []factory.Informer) {
+	hook := csidrivercontrollerservicecontroller.WithConfigMapHashAnnotationHook(
+		c.ControlPlaneNamespace,
+		cinderConfigName,
+		c.GetControlPlaneConfigMapInformer(c.ControlPlaneNamespace),
+	)
+	informers := []factory.Informer{}
+	return hook, informers
+}
+
+// withConfigDeploymentHook adds annotations based on the hash of the config map containing our config,
+// ensuring we restart if that changes
+func withConfigDaemonSetHook(c *clients.Clients) (csidrivernodeservicecontroller.DaemonSetHookFunc, []factory.Informer) {
+	hook := csidrivernodeservicecontroller.WithConfigMapHashAnnotationHook(
+		c.ControlPlaneNamespace,
+		cinderConfigName,
+		c.GetControlPlaneConfigMapInformer(c.ControlPlaneNamespace),
+	)
+	informers := []factory.Informer{}
+	return hook, informers
+}
+
+func createConfigMapSyncer(c *clients.Clients) (factory.Controller, error) {
```

Comment:
add a comment about what we are syncing and why.

Reply:
Done